src/types/data/text.js

```diff
@@ -24,6 +24,7 @@ export default class TextType extends DataType {
     url: /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/,
     ssn: /^(\d{3}-?\d{2}-?\d{4}|XXX-XX-XXXX)$/,
     ssnDisplay: /^(\d{3}-\d{2}-\d{4})$/,
```

Comment:
This display SSN should also check for `XXX-XX-XXXX`.  This value is used when there is no SSN applicable in this situation.  Look at how the `ssn` regex on the line above checks for it for an example.